vmm/src/device_manager.rs

```diff
@@ -2352,14 +2352,21 @@ impl DeviceManager {
 
     #[cfg(feature = "pci_support")]
     fn create_kvm_device(vm: &Arc<dyn hypervisor::Vm>) -> DeviceManagerResult<DeviceFd> {
-        let mut vfio_dev = hypervisor::kvm::kvm_create_device {
-            type_: hypervisor::kvm::kvm_device_type_KVM_DEV_TYPE_VFIO,
-            fd: 0,
-            flags: 0,
-        };
+        #[cfg(feature = "kvm")]
+        {
+            let mut vfio_dev = hypervisor::kvm::kvm_create_device {
+                type_: hypervisor::kvm::kvm_device_type_KVM_DEV_TYPE_VFIO,
+                fd: 0,
+                flags: 0,
+            };
```

Comment:
I think `create_kvm_device()` should be removed from device_manager.

What it's trying to do here is to ask the hypervisor crate to create a specific kind of device, a passthrough one. It happens to be so that with KVM being a Linux API, passthrough practically means VFIO.

So I think it would be cleaner to expand the hypevisor's `vm` trait with a `create_passthrough_device(&self)` method and abstract the "KVM means VFIO" dependency in the hypervisor KVM implementation:

```Rust
    fn create_passthrough_device(&self) -> vm::Result<DeviceFd> {
        let mut vfio_dev = kvm_create_device {
            type_: kvm_device_type_KVM_DEV_TYPE_VFIO,
            fd: 0,
            flags: 0,
        };

        self.create_device(&mut vfio_dev)
    }
```

Then any hypervisor that does not support passthrough would return an error for that method.

Reply:
OK. Let me experiment with this idea a bit.